code/datums/controllers/explosion_controls.dm

```diff
@@ -141,18 +141,21 @@ var/datum/explosion_controller/explosions
 	proc/process()
 		if (exploding)
 			return
-		else if (queued_turfs.len)
+		else if (length(queued_turfs))
 			kaboom()
 
-		if (queued_explosions.len)
+		if (length(queued_explosions))
 			var/datum/explosion/E
-			while (queued_explosions.len)
+			while (length(queued_explosions))
 				E = queued_explosions[1]
 				queued_explosions -= E
 				E.explode()
 				next_turf_safe |= E.turf_safe
 
 
+/obj
+	var/datum/explosion/last_explosion = null
```

Comment:
please clear this in disposing

Reply:
Hopefully this is the right way to do that?